dashboard/app/models/user.rb

```diff
@@ -246,6 +247,19 @@ def delete_course_as_facilitator(course)
     courses_as_facilitator.find_by(course: course).try(:destroy)
   end
 
+  def log_admin_save
+    # Do not log for development or adhoc environments.
+    return unless [:test, :staging, :levelbuilder, :production].include? rack_env
+
+    ChatClient.message 'infra-security',
+      "#{admin ? 'Granting' : 'Revoking'} UserPermission: "\
+      "environment: #{rack_env}, "\
+      "user ID: #{id}, "\
+      "email: #{email}, "\
+      "permission: ADMIN",
+      color: 'red'
```

Comment:
It appears that green, red, yellow, and purple are the recommended options, so I'll go with yellow: https://github.com/code-dot-org/code-dot-org/blob/bb14e5e42e007ac868e209953a8b6e7a686f4423/lib/cdo/slack.rb#L5-L10